### PR TITLE
[TTAHUB-304] Recipients with class scores with goals details

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

I have not hooked up some of the functionality on this component as I don't think we will be able to get away with using the 'usePagingWidget' hook. For the purposes of this PR we can simply review the UI design.

## How to test

- Review the code (noting that some stuff isnt hooked up)
- Make sure we match the design.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3304

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/hooks/useWidgetPaging.js [hunk 1]
```diff
@@ -126,6 +126,21 @@ export default function useWidgetPaging(
...
+          heading: d.name,
+          data: Object.keys(d)
+            .filter((key) => headers.includes(key))
+            .map((key) => ({
+              title: key,
+              value: d[key].toString().replace(/,/g, ''),
+            })),
+        }));
+      }
+
```
Comment (thewatermethod) [inline frontend/src/hooks/useWidgetPaging.js:143]: can you explain in a comment why we need this? I don't get it by looking at it.

File: frontend/src/pages/QADashboard/Components/GoalCard.scss
Comment (thewatermethod) [inline frontend/src/pages/QADashboard/Components/GoalCard.scss:1]: empty file, can we remove?

File: frontend/src/pages/QADashboard/Components/RecipientCard.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+              label=""
+              value={recipient.id}
+              checked={isChecked}
+              onChange={handleGoalCheckboxSelect}
+              aria-label={`Select recipient ${recipient.name}`}
+              className="margin-right-1"
+              data-testid="selectRecipientTestId"
+            />
+          </div>
+          <div className="ttahub-recipient-card__recipient-column__break" />
```
Comment (thewatermethod) [inline frontend/src/pages/QADashboard/Components/RecipientCard.js:44]: why do we need an empty element?

File: frontend/src/hooks/useWidgetPaging.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/hooks/useWidgetPaging.js:143]: Yea this probably will go away. I was attempting to re-use the export to CSV for a different underlying set. But given the fact that this data displays totally differently I probably will write a separate export to CSV (or handle this on the BE).

In short this code will probably go away before being merged into main.

File: frontend/src/pages/QADashboard/Components/GoalCard.scss
Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/Components/GoalCard.scss:1]: removed.

File: frontend/src/pages/QADashboard/Components/RecipientCard.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/Components/RecipientCard.js:44]: This fixes an issue when the window is shrunk shown below (let me know if you have time to go over a better solution):

Good
image

Bad
image

Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/Components/RecipientCard.js:44]: Neverminded this has been removed.

**Comment by thewatermethod** [comment]
Screenshot 2024-09-23 at 9 26 35 AM

Still seeing issues on these. In the screenshot, I've fixed the goal number.

Here is the code for a suggested fix (we can wrap each row in a component so that it's easier to maintain and use in the future)

```jsx
const DataRow = ({ label, value }) => (
  <Grid row className="margin-bottom-1">
    <Grid
      desktop={{ col: 2 }}
      mobileLg={{ col: 12 }}
      gap={{ desktop: 2 }}
    >
      {label}
    </Grid>
    <Grid
      desktop={{ col: 10 }}
      mobileLg={{ col: 12 }}
    >
      {value}
    </Grid>
  </Grid>
);

...
// example usage
        <DataRow
          label="Goal number"
          value={(
            <Link to="recipient-tta-records/376/region/1/goals?id[]=83697">
              {goal.goalNumber}
            </Link>
          )}
        />
```

**Comment by thewatermethod** [comment]
Screenshot 2024-09-23 at 9 40 44 AM

Alignment here doesn't match the mockup either

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**